pkg/postage/batchstore/store.go

```diff
@@ -97,6 +100,15 @@ func (s *store) GetReserveState() *postage.ReserveState {
 	}
 }
 
+func (s *store) IsWithinStorageRadius(addr swarm.Address) bool {
+
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+
+	po := swarm.Proximity(addr.Bytes(), s.base.Bytes())
+	return po >= s.GetReserveState().StorageRadius
```

Comment:
Maybe its better to create a new interface `StorageRadiusGetter` and implement it in the batchstore. Its being used at a lot of places and maybe having a separate interface would be more flexible for eg in tests etc.

Also it would be better not to expose unnecessary reserve info to other packages.